src/lib/server.ts

```diff
@@ -411,24 +413,37 @@ export class ZwavejsServer extends EventEmitter {
 
   constructor(
     private driver: Driver,
-    private options: ZwavejsServerOptions = {}
+    private options: ZwavejsServerOptions = {},
+    private destroyServerOnHardReset: boolean = false
   ) {
     super();
     this.logger = options.logger ?? console;
-    this.driver.updateUserAgent({ [applicationName]: version });
   }
 
   async start() {
     if (!this.driver.ready) {
       throw new Error("Cannot start server when driver not ready");
     }
 
+    this.driver.updateUserAgent({ [applicationName]: version });
     this.server = createServer();
     this.wsServer = new ws.Server({
       server: this.server,
       perMessageDeflate: true,
     });
     this.sockets = new ClientsController(this.driver, this.logger);
+    this.sockets.on("hard reset", async () => {
```

Comment:
this event should be called `perform hard reset`. All other events are named after something that happened, ie `driver ready`. But in this case the server needs to be reset.

_It also feel a little weird that this is using the event system._

Reply:
that's fair. I could make it an async function on the server instance and pass the instance up to the handler. That would require keeping a reference to the server in the ClientController and Client, but then the driver would handle all of the logic. Thoughts?

Reply:
I don't have the architecture in my head atm. Do we have an architecture diagram I can reference ?

Reply:
we don't but I can try to create one later this week

Reply:
does this help? generated from the `classdiagram-ts` VSC extension:
<img width="441" alt="image" src="https://user-images.githubusercontent.com/7243222/205818697-8a912596-dca7-43e0-be7d-6a2c8203b2da.png">

Reply:
seems passing a reference to server to ClientController is ok.